test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,325 @@
+        Scrollbar scrollbar = new Scrollbar(Scrollbar.HORIZONTAL, 0, 1, 0, 255);
+        TextField textfield = new TextField(15);
+        TextArea textarea = new TextArea(5, 15);
+
+        components = new Component[] { panel, button, label, list, choice,
+            checkbox, scrollbar, textfield, textarea, frame };
+
+        for (int i = 0; i < components.length - 1; i++) {
+            components[i].addComponentListener(componentListener);
+            frame.add(components[i]);
+        }
+        frame.addComponentListener(componentListener);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(ComponentEventTest::initializeGUI);
+            robot.waitForIdle();
+            robot.delay(DELAY);
+
+            doTest();
+
+            System.out.println("Test PASSED");
+        } finally {
+            EventQueue.invokeAndWait(ComponentEventTest::disposeFrame);
+        }
+    }
+
+    private static void doTest()
+        throws InvocationTargetException, InterruptedException {
+
+        EventQueue.invokeAndWait(() -> {
+            compAt = components[0].getLocationOnScreen();
+            compSize = components[0].getSize();
+        });
+
+        robot.mouseMove(compAt.x + compSize.width / 2,
+            compAt.y + compSize.height / 2);
+
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+        for (int i = 0; i < components.length; i++) {
+            for (boolean state : new boolean[] { true, false }) {
+                doTest(i, state);
+            }
+        }
+
+        robot.delay(DELAY);
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            frame.setExtendedState(Frame.ICONIFIED);
+        });
+
+        robot.delay(DELAY);
+        if (componentShown || componentHidden || componentMoved
+            || componentResized) {
+            printErrEvents("ComponentEvent triggered when frame is iconified");
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            frame.setExtendedState(Frame.NORMAL);
+        });
+
+        robot.delay(DELAY);
+        if (componentShown || componentHidden) {
+            printErrEvents("ComponentEvent triggered when frame set to normal");
+        }
+    }
+
+    private static void printErrEvents(String errorMsg) {
+        System.err.print("Events triggered are: ");
+        for (int j = 0; j < events.size(); j++) {
+            System.err.print(events.get(j) + "; ");
+        }
+        System.err.println();
+        throw new RuntimeException("FAIL: " + errorMsg);
+    }
+
+    private static void doTest(int i, boolean enable)
+        throws InvocationTargetException, InterruptedException {
```

Comment:
I think you should print the current value of the `enabled` flag. If the test fails for any component, you want to know if it fails when components are enabled or disabled.